[ChongCheeKaiClarence] iP

ChongCheeKaiClarence opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes)) 

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use
- 
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

-[x] Managing tasks
-[x] Managing deadlines (coming soon)
-[x] Reminders (coming soon)

___

If you Java programmer, you can use it to practice Java too. Here's the `main` method:




```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Added level 1 work

Commit:
Added Level 2

Commit:
Added task.java and Level 3 work

Commit:
Added Deadline.java, Event.java, Todo.class and Level 4 work

Commit:
edited Input.txt and Expected.txt. A-TextUiTesting done

Commit:
Added level 5 work

Commit:
Added level 6 work

Commit:
Added enums.
Changed names from Deadline, Event and Todo to DeadlineTask, EventTask and TodoTask.
Added javadocs.

Commit:
Add checkstyle

Commit:
Add Level 7

Commit:
Implement Level 7 with runtest

Commit:
Add Level 8 in branch

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Change save file Strings to match Level 8, add more exception

Commit:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

* 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip:
  Add Gradle support

Commit:
Add Use More OOP

Commit:
Add packages

Commit:
Add JUnit Tests

Commit:
Create a JAR File

Commit:
Add JavaDoc comments

Commit:
Follow the Coding Standard

Commit:
Add find

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DeadlineTask.java
#	src/main/java/duke/TaskList.java

Commit:
Merge branch 'branch-Level-9'

Commit:
Add coding standard

rexong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+        this.ui = new Ui(this.parser);
+        try {
+           this.parser.load();
+            System.out.println("Hello! I'm Duke\nWhat can I do for you?\n" );
```
The indentation here seems to be off.

rexong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+ */
+public class Parser {
+
+	private static Pattern CHECKSTRING = Pattern.compile("-?\\d+");
```
Should `CHECKSTRING` be in all caps? I think this is a static variable and thus should use a camelCase convention like `checkString`.

Jiacheng-y replied:
Alternatively, perhaps `final` can be added as a keyword if you had meant for CHECKSTRING to be a constant. 
```java
private static final Pattern CHECKSTRING =Pattern.compile("-?\\d+");
```

rexong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+		String[] parts = response.split(" ", 2);
+		/** the action */
+		String part1 = parts[0];
+		String part2;
```
Perhaps the variables' name could be more intuitive? Instead of using `part1` and `part2`.

rexong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,125 @@
+ */
+public class Storage {
+
+	private final String filePath;
```
Should `filePath` be in camelCase? I think because of the keyword `final`, `filePath` should be named `FILE_PATH` instead.

ChongCheeKaiClarence replied:
filePath does not have a static modifier. So I do not think it should be in caps as it is not a constant.

rexong commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,47 @@
+
+	private Parser parser;
+
+	private static final Scanner echo = new Scanner(System.in);
```
Should this be `ECHO` instead? Since it is a constant.

rexong commented on src/test/java/duke/DeadlineTaskTest.java:
```diff
@@ -0,0 +1,14 @@
+package duke;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Shouldn't the static import come before the regular import? 
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
```

rexong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+        try {
+           this.parser.load();
+            System.out.println("Hello! I'm Duke\nWhat can I do for you?\n" );
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        } catch (DateTimeParseException e) {
+            System.out.println("Date and Time format error: check input or file, " + e.getMessage());
+        } catch (DukeException e) {
+            throw new RuntimeException(e.getMessage());
+        }
```
I like how you specify each exception. It makes it easier to read and follow. 😃

rexong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+	private Storage storage;
+	private TaskList tasks;
+
+	enum Update {
```
I feel that `Update` could perhaps be changed into `TaskUpdater` instead. Converting it from a verb to a noun.

rexong reviewed (commented):
Your code looks good! 👍 You followed most of the coding standards stated in [here](https://se-education.org/guides/conventions/java/intermediate.html#naming). Just some minor tweaks! 😄

Commit:
Add Level 10

Commit:
clean up code

Commit:
Fix bug where tasks from txt file does not give the right mark symbol

Commit:
Merge branch 'Correct-error-in-text-ui-branch'

# Conflicts:
#	data/tasks.txt

Commit:
Fix bug where GUI does not show the full list

Commit:
Fix bug where MainWindow was set to java 18 instead of java 11.0.2

Commit:
Merge branch 'branch-Level-10'

Commit:
Add CheckStyle

Commit:
Edit javadocs

Commit:
Merge branch 'branch-CheckStyke'

Jiacheng-y commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+		case "bye":
+			System.out.println("-----------------------------------------------");
+			System.out.println("Bye. Hope to see you again soon!");
+			System.out.println("-----------------------------------------------");
```
Perhaps these few lines can be extracted out into the Ui.java class for a better segregation of functionality across classes.

Jiacheng-y commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+ */
+public class TaskList {
+
+	private static Pattern CHECKSTRING = Pattern.compile("-?\\d+");
```
same issue as above in line 12 of Parser.java

Jiacheng-y commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+		System.out.println("-----------------------------------------------");
+		System.out.println("Nice! I've marked this task as done:");
+		System.out.println(task);
+		System.out.println("-----------------------------------------------");
```
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.

Jiacheng-y commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+		System.out.println("-----------------------------------------------");
+		System.out.println("OK, I've marked this task as not done yet:");
+		System.out.println(task);
+		System.out.println("-----------------------------------------------");
```
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.

Jiacheng-y commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+		System.out.println("Got it. I've added this task:");
+		System.out.println(task);
+		System.out.println("Now you have " + this.list.size() +" tasks in the list.");
+		System.out.println("-----------------------------------------------");
```
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.

Jiacheng-y commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+		System.out.println("Noted. I've removed this task:");
+		System.out.println(temp);
+		System.out.println("Now you have " + this.list.size() +" tasks in the list.");
+		System.out.println("-----------------------------------------------");
```
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.

Jiacheng-y reviewed (commented):
Generally looks good to merge, just a few coding standards and code quality issues to relook. The way you've written some of your functions were quite interesting!

Commit:
There is no code to handle null.

There is a possibility to getting a null in the parameters.

Add  Assertion methods.

To allow us to handle errors and know when a null is produced.

Commit:
As of now, assumptions are checked with if statements
or throwing exceptions.

It is problematic if the assumption does not hold and could
possibly crash the system.

Add Assertions

This will allow our assumption to be checked before proceeding on
with running the code.

Assertions ensures that my code fulfils the assumptions I made.

Commit:
As of now, the Code Quality guidelines are not met.

It is confusing to read for other developers.

Improve Code Quality.

The code is now more readable.

Now, the code follows the Code quality guidelines.

ChongCheeKaiClarence committed:
Merge pull request #1 from ChongCheeKaiClarence/branch-A-Assertions

Add A-Assertions

Commit:
Merge branch 'master' of https://github.com/ChongCheeKaiClarence/ip

* 'master' of https://github.com/ChongCheeKaiClarence/ip:
  As of now, assumptions are checked with if statements or throwing exceptions.
  There is no code to handle null.

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/DeadlineTask.java
#	src/main/java/duke/EventTask.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

ChongCheeKaiClarence committed:
Merge pull request #3 from ChongCheeKaiClarence/branch-A-CodeQuality

Improve Code Quality

Commit:
Merge branch 'master' of https://github.com/ChongCheeKaiClarence/ip into branch-A-CodeQuality

* 'master' of https://github.com/ChongCheeKaiClarence/ip:

Commit:
Merge branch 'master' of https://github.com/ChongCheeKaiClarence/ip

* 'master' of https://github.com/ChongCheeKaiClarence/ip:
  As of now, the Code Quality guidelines are not met.

Commit:
As of now, there is no way of categorizing the tasks.

It is hard to associate them to each other.

Add C-Tagging.

Now, we can see how the tasks are associated to each other.

We can use these tags for relevent find and sort.

ChongCheeKaiClarence committed:
Merge pull request #4 from ChongCheeKaiClarence/branch-BCD-Extension

Add C-Tagging

Commit:
Merge branch 'master' of https://github.com/ChongCheeKaiClarence/ip

* 'master' of https://github.com/ChongCheeKaiClarence/ip:
  As of now, there is no way of categorizing the tasks.

ChongCheeKaiClarence committed:
Merge pull request #5 from ChongCheeKaiClarence/branch-A-CodeQuality

Merge branch 'master' of https://github.com/ChongCheeKaiClarence/ip i…

Commit:
Add A-CI

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Remove IO redirection from gradle.yml and Remove print
statement in Duke.java

Commit:
Update testcases

Commit:
Improve tag viewing
Add user guide

Commit:
Fix README, image not rendering

Commit:
Change png to remove excess border

Commit:
Update user guide

Commit:
Improve code quality

Commit:
Solve checkstyle error

Commit:
Improve code quality by adding method header comments.

Commit:
Cut down Jar size by removing javafx block in build.gradle

Commit:
Improve code quality

Commit:
Add window resizing

Commit:
Fix array index exception
Add javafx block to build.gradle

Commit:
Shift javafx block above the dependencies block
